x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -360,8 +371,24 @@ export const statusCheckAlertFactory: UptimeAlertTypeFactory<ActionGroupIds> = (
           ...updateState(state, true),
         });
 
+        const relativeViewInAppUrl = getMonitorRouteFromMonitorId({
+          monitorId: monitorSummary.monitorId,
+          dateRangeEnd: 'now',
+          dateRangeStart: indexedStartedAt,
+          filters: {
+            'observer.geo.name': [monitorSummary.observerLocation],
+          },
+        });
+        const viewInAppUrl = basePath.publicBaseUrl
+          ? new URL(
+              join(basePath.serverBasePath, relativeViewInAppUrl),
+              basePath.publicBaseUrl
+            ).toString()
+          : relativeViewInAppUrl;
```

Comment:
basePath can be present even if publicBaseUrl isn't defined. Does it even makes sense to define this var if publicBaseUrl isn't defined? 

if not defined then we can link to docs explaninig that you need to define this.

though yes i know, on cloud this will always be available.

Reply:
@shahzad31, you are right, `publicBaseUrl` on our Cloud solution it's always there, I put this for all other solutions e.g. on-premise.

In all cases `viewInAppUrl ` is required, but I need to check the presence of the `publicBaseUrl ` to either build the URL as a full one or using the relative one (without domain name)